tests/components/forked_daapd/test_media_player.py

```diff
@@ -0,0 +1,726 @@
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_TOGGLE)
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_VOLUME_MUTE,
+        {ATTR_MEDIA_VOLUME_MUTED: True},
+    )
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_VOLUME_MUTE,
+        {ATTR_MEDIA_VOLUME_MUTED: False},
+    )
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_VOLUME_SET,
+        {ATTR_MEDIA_VOLUME_LEVEL: 0.5},
+    )
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_MEDIA_PAUSE)
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_MEDIA_PLAY)
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_MEDIA_STOP)
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_MEDIA_PREVIOUS_TRACK)
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_MEDIA_NEXT_TRACK)
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_MEDIA_SEEK,
+        {ATTR_MEDIA_SEEK_POSITION: 35},
+    )
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_CLEAR_PLAYLIST)
+    await _service_call(
+        hass, TEST_MASTER_ENTITY_NAME, SERVICE_SHUFFLE_SET, {ATTR_MEDIA_SHUFFLE: False}
+    )
+    # stop player and run more stuff
+    state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert state.attributes[ATTR_MEDIA_VOLUME_LEVEL] == 0.2
+    get_request_return_values["player"] = SAMPLE_PLAYER_STOPPED
+    updater_update = mock_api_object.start_websocket_handler.call_args[0][2]
+    await updater_update(["player"])
+    await hass.async_block_till_done()
+    # mute from volume==0
+    state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert state.attributes[ATTR_MEDIA_VOLUME_LEVEL] == 0
+    await _service_call(
+        hass,
+        TEST_MASTER_ENTITY_NAME,
+        SERVICE_VOLUME_MUTE,
+        {ATTR_MEDIA_VOLUME_MUTED: True},
+    )
+    # now turn off (stopped and all outputs unselected)
+    get_request_return_values["outputs"] = SAMPLE_OUTPUTS_UNSELECTED
+    await updater_update(["outputs"])
+    await hass.async_block_till_done()
+    # toggle from off
+    await _service_call(hass, TEST_MASTER_ENTITY_NAME, SERVICE_TOGGLE)
+    for output in SAMPLE_OUTPUTS_ON:
+        mock_api_object.change_output.assert_any_call(
+            output["id"], selected=output["selected"], volume=output["volume"],
+        )
+    mock_api_object.set_volume.assert_any_call(volume=0)
+    mock_api_object.set_volume.assert_any_call(volume=SAMPLE_PLAYER_PAUSED["volume"])
+    mock_api_object.set_volume.assert_any_call(volume=50)
+    mock_api_object.set_enabled_outputs.assert_any_call(
+        [output["id"] for output in SAMPLE_OUTPUTS_ON]
+    )
+    mock_api_object.set_enabled_outputs.assert_any_call([])
+    mock_api_object.start_playback.assert_called_once()
+    assert mock_api_object.pause_playback.call_count == 3
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)